cli/azd/pkg/ioc/container.go

```diff
@@ -0,0 +1,92 @@
+package ioc
```

Comment:
Can you add a brief description on what should go into this package long-term? If I understand correctly, it's a wrapper for `golobby/container` to satisfy our needs + container hierarchies.